benchmarks/test_elemwise.py

```diff
@@ -0,0 +1,55 @@
+import importlib
+import operator
+import os
+
+import sparse
+
+import pytest
+
+import numpy as np
+import scipy.sparse as sps
+
+DENSITY = 0.001
+
+
+def get_test_id(side):
+    return f"{side=}"
+
+
+@pytest.fixture(params=[100, 500, 1000], ids=get_test_id)
+def elemwise_args(request, seed, max_size):
+    side = request.param
+    if side**2 >= max_size:
+        pytest.skip()
+    rng = np.random.default_rng(seed=seed)
+    s1_sps = sps.random(side, side, format="csr", density=DENSITY, random_state=rng) * 10
+    s1_sps.sum_duplicates()
+    s2_sps = sps.random(side, side, format="csr", density=DENSITY, random_state=rng) * 10
+    s2_sps.sum_duplicates()
+    return s1_sps, s2_sps
+
+
+@pytest.mark.parametrize("f", [operator.add, operator.mul, operator.ge])
+@pytest.mark.parametrize("backend", ["SciPy", "Numba", "Finch"])
+def test_elemwise(benchmark, f, backend, elemwise_args):
+    if backend in ["Numba", "Finch"]:
+        os.environ[sparse._ENV_VAR_NAME] = backend
+        importlib.reload(sparse)
```

Comment:
This could end up permanently corrupting state. Let's define a backend fixture:

```python
@pytest.fixture(scope="function")
def backend(params):
    if backend in {"Numba", "Finch"}:
        os.environ[sparse._ENV_VAR_NAME] = backend
        importlib.reload(sparse)
    yield some_kind_of_conversion_function, sparse
    del os.environ[sparse._ENV_VAR_NAME]
    importlib.reload(sparse)
```

The part after the `yield` will make sure to "fix" the state after it's been modified, so it doesn't affect other tests.

And use it as

```python
def test_fn(backend, arr_sps):
    convert, sparse = backend
    arr = convert(arr_sps)
    ...
```

Reply:
Thanks..